Source/HTTP/iOS/http_ios.mm

```diff
@@ -0,0 +1,143 @@
+// Copyright (c) Microsoft Corporation
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+#include "pch.h"
+#include "http_ios.h"
+
+NAMESPACE_XBOX_HTTP_CLIENT_BEGIN
+
+ios_http::ios_http(_In_ AsyncBlock* asyncBlock, _In_ hc_call_handle_t call) :
+    m_call(call),
+    m_asyncBlock(asyncBlock),
+    m_sessionTask(nil)
+{
+    NSURLSessionConfiguration* configuration = NSURLSessionConfiguration.ephemeralSessionConfiguration;
+
+    uint32_t timeoutInSeconds = 0;
+    if (FAILED(HCHttpCallRequestGetTimeout(m_call, &timeoutInSeconds)))
+    {
+        // default to 60 to match other default ios behaviour
+        timeoutInSeconds = 60;
+    }
+    
+    [configuration setTimeoutIntervalForRequest:(NSTimeInterval)timeoutInSeconds];
+    [configuration setTimeoutIntervalForResource:(NSTimeInterval)timeoutInSeconds];
+    
+    m_session = [NSURLSession sessionWithConfiguration:configuration];
+}
+
+void ios_http::completion_handler(NSData* data, NSURLResponse* response, NSError* error)
+{
+    assert([response isKindOfClass:[NSHTTPURLResponse class]]);
+    NSHTTPURLResponse* httpResponse = (NSHTTPURLResponse*)response;
+    
+    if (error)
+    {
+        uint32_t errorCode = static_cast<uint32_t>([error code]);
+        HCHttpCallResponseSetNetworkErrorCode(m_call, errorCode, errorCode);
```

Comment:
>errorCode [](start = 54, length = 9)

this should still be E_FAIL

the error code from iOS is not an HRESULT and should not be used as one